### ISSUE TITLE
Support definining linters from startup files

### ISSUE THREAD

**Issue by w0rp** [w0rp/ale#1278]
Currently only defining linters in `ale_linters` directories in `runtimepath` is supported. It wouldn't be too much work to add support for defining linters anywhere. To implement this, a separate Dictionary should be used in `autoload/ale/linter.vim` for tracking if the `runtime` line has been run already or not. Then it will be possible to add on linters before or after searching `runtimepath`.

The code should probably be changed so linters can be predictably replaced by the linter name, rather than letting you define two linters with the same name.

**Comment by w0rp** [comment]
I think what would make the most sense would be to have `Define` replace an existing linter with the same name by default, and then support an optional setting for preventing linters from being replaced, like `'priority': 1`.

**Comment by languitar** [comment]
Sounds good to me

**Comment by w0rp** [comment]
Now linters can be defined in pretty much any Vim file, as long as the `Define` function can be loaded itself. I decided not to implement priority yet, until it becomes a problem for someone. The last linter loaded will win if two have the same name, so that will be `runtimepath` over `vimrc`, and anything after `runtimepath` over `runtimepath`. I added a new function `ale#linter#PreventLoading(filetype)` to disable loading any linters from `runtimepath` for a given filetype, and calling `ale#linter#Get(filetype)` before `ale#linter#Define` will let you replace a runtimepath linter of the same name.
